Pass JSON config to supervise on stdin

kendru opened the pull request:
Fixes #47

brandonbloom commented on cmd/supervise/main.go:
```diff
@@ -1,13 +1,21 @@
-	args := os.Args[1:]
-	supervise.Main(command, args)
+	cfg := &supervise.Config{}
+	if err := json.NewDecoder(os.Stdin).Decode(cfg); err != nil {
```
why duplicate this code rather than stick it inside `supervise.Main`?

kendru replied:
I don't want supervise.Main to depend on reading from `os.Stdin`, and it is called from exop and exod, which do not expect input on stdin.

brandonbloom replied:
I'm confused about this. Seems like there is some leftover dev stuff that is marked "XXX" and "This is broken". Should we remove/cleanup that code?

Regarding `exop` - I'm not sure why that code would need a `supervise` codepath at all. I don't remember adding that. There is already the `supervise/main.go` codepath for testing the supervise script directly without the exo binary. Integrating it in to exop seems redundant.

As for the exod codepath, I think that was a quick and dirty hack to deal with the fact that exod wants to launch supervise by looking for `argv[0]` to spawn a new process. Now that we have a proper config mechanism, maybe we should use that instead to supply a supervise path?

brandonbloom commented on internal/providers/unix/components/process/process.go:
```diff
@@ -47,30 +48,24 @@ func (p *Process) start(ctx context.Context) error {
 	}
 
+	// Pipe JSON config to supervise on stdin.
+	configJSON := supervise.MustEncodeConfig(&supervise.Config{
```
I don't see this function defined anywhere...

kendru replied:
I just added the file that *should* have been there initially. Thanks for catching this!

brandonbloom reviewed (changes requested):
I _think_ this PR is missing a file or something for the MustEncodeConfig function?

brandonbloom commented on cmd/exop/main.go:
```diff
@@ -30,15 +29,24 @@ func main() {
 	// subcommand so that it always uses the latest veresion of the code rather than relying
```
typo: veresion

brandonbloom reviewed (approved):
LGTM. I might take the opportunity to do one more cleanup step improving, removing or otherwise commenting the various entrypoints. Up to you.

kendru commented:
@brandonbloom I removed the entrypoints to supervise from `exop` and `exod`, and I removed the top-level `supervise` command entirely. If you use exop and exod and need something from them, let me know and I can add it.
